addons/binding/org.openhab.binding.kostalpikoiqplenticore/README.md

```diff
@@ -0,0 +1,2964 @@
+# kostalpikoiqplenticore Binding
+
+
+This binding allows you to communicate with KOSTAL solar inverters.
+Currently the required communication board is embedded in PIKO IQ and PLENTICORE plus inverters.
+
+The communication board (SCB = Smart Communication Board) offers a (not yet officially released, but stable) web API to consume all information, which are available in the inverter´s own web front end.
+This includes current operating data as well as aggregated data from the past for statistical purposes. 
+
+
+## Supported Things
+
+This implementation was tested for the current KOSTAL PIKO PLENTICORE plus and PIKO IQ devices.
+All of these devices contain the same communication board (SCB = <u>S</u>mart<u>C</u>onnection<u>B</u>oard)
+
+Currently supported things are:
+<ul>
+<li>PIKO IQ 4.2</li>
+<li>PIKO IQ 5.5</li>
+<li>PIKO IQ 7.0</li>
+<li>PIKO IQ 8.5</li>
+<li>PIKO IQ 10.0</li>
+<li>PLENTICORE plus 4.2 (with or without battery attached)</li>
+<li>PLENTICORE plus 5.5 (with or without battery attached)</li>
+<li>PLENTICORE plus 7.0 (with or without battery attached)</li>
+<li>PLENTICORE plus 8.5 (with or without battery attached)</li>
+<li>PLENTICORE plus 10.0 (with or without battery attached)</li>
+</ul>
+
+Others may be supported (like future devices using the same SCB or offering the same web api, branded oem devices, ...), but they were not tested!
```

Comment:
Naming it was pretty hard … :-)

You are totaly right with the "future proof" thing, I had that same thought
Some facts that made it so hard to choose a name:
1) This is currently the 3. Generation of KOSTAL inverters and the first one with Linux based communication board. Don´t  know if future devises will support this API
2) Maybe there will be other devices based on this board or not... 
3) API may change, but still be a web API (just incompatible)
4) KOSTAL also manufactures devices for other OEMs … this devices are also compatible with the addon, since it´s the same software. Unfortunatly I don´t know the brands 
5) Choosing a brand-Independent Name could lead to users not finding the plugin in the repository :(

After all this thoughts I stuck with the Information I know: the productnames of the KOSTAL inverters.

Maybe you have an idea or there are similar cases in the openHAB repository. where some basic decision was made to which I can align my naming?

Reply:
Any reason you didn't add this to the `kostalinverter` binding? It seems logical as they are from the same company, same type of devices, only a different way to connect. Then the user also doesn't have to decide what binding to install. Especially since the kostalinverter binding doesn't mention anything about versions it's difficult for the user to see what binding is needed. Some improvements can be made in that area.

Reply:
It would certainly solve the bundle naming issue and confusion that users don't know which binding to install.

The whole idea of open source is to mess around with the work of others. :wink: 
If the code for all generations is part of one binding it is best to also adopt the same naming standards for channels and thing types. What are your thoughts on this @cschneider?

Perhaps the code can be structured in packages that specifying the generations (gen1, gen2, etc.)?

Another approach could be to structure code for different generations in different bundles and install them in the same feature. I.e. similar to how the bluetooth binding is structured. What preference do you have for this @kaikreuzer?

Reply:
I merged the addon to the existing one :)
Therefor I consider the naming question as resolved